### PR TITLE
add proper .gitignore to cleanup git status

### PR DESCRIPTION
**Pull request by sitano**

### CODE REVIEW

**Comment by glommer** [comment]
Thank you! I think your subject got broken, though. "to cleanup git st". What is st?

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.idea
+*.iml
+/target
+Cargo.lock
```
Comment (glommer) [inline .gitignore:4]: Cargo.lock is indeed pretty important and target is good too.

The others I don't mind, it seems like some IDE stuff, right?

**Review by glommer — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (sitano) [reply .gitignore:4]: > Cargo.lock is indeed pretty important and target is good too.

Do you mean that `/target` and `cargo.lock` must not be in .gitignore? Or you agree?

Others are IntelliJ project files.

**Comment by sitano** [comment]
> Thank you! I think your subject got broken, though. "to cleanup git st". What is st?

`git st` is an alias for `git status`

File: .gitignore (hunk 1 above)
Comment (sitano) [reply .gitignore:4]: I mean you are never going to commit /target and lock into the repo, right? Then they must be hidden from the git status.

Comment (glommer) [reply .gitignore:4]: I was just voicing agreement, sorry if I confused you.

As far as I know we can add files from various IDEs here too, so I am cool with that.

**Review by glommer — APPROVED** [review]

**Pull request merged**